internal/go/skycfg/assert.go

```diff
@@ -113,9 +113,10 @@ func (t *TestContext) CallInternal(thread *starlark.Thread, args starlark.Tuple,
 
 	if !val {
 		buf := new(strings.Builder)
-		thread.Caller().WriteBacktrace(buf)
+		stk := thread.CallFrame(1)
+		fmt.Fprintf(buf, "%s", stk)
 		err := assertionError{
-			position:  thread.Caller().Position().String(),
+			position:  stk.Pos.String(),
```

Comment:
Rather than duplicating the stack trace formatting logic, the frame could be stored in `assertionError` and the formatting handled in `assertionError.Error()`. Something like this:
```go
err := assertionError{
	stack: thread.CallStack(),
}
```